src/activation.rs

```diff
@@ -0,0 +1,180 @@
+use wayland_client::{
+    globals::{BindError, GlobalList},
+    protocol::{wl_seat, wl_surface},
+    Dispatch, Proxy, QueueHandle,
+};
+use wayland_protocols::xdg::activation::v1::client::{xdg_activation_token_v1, xdg_activation_v1};
+
+use crate::{
+    error::GlobalError,
+    globals::{GlobalData, ProvidesBoundGlobal},
+};
+
+#[derive(Debug, Clone)]
+pub struct RequestData {
+    pub app_id: Option<String>,
+    pub seat_and_serial: Option<(wl_seat::WlSeat, u32)>,
+    pub surface: Option<wl_surface::WlSurface>,
+}
```

Comment:
I think it should point to docs on at least `RequestDataExt`, otherwise how the user would know what the fields should do?

Reply:
Docs updated